### ISSUE TITLE
Decoding mesh with TEX_COORD attribute fails after encoding with specific speed/quantization config 

### ISSUE THREAD

**Issue by neonglb** [google/draco#800]
Hi,
the following code snippet exemplifies my observation that decoding fails after setting speed<=3 and defining quantization_bits of TEX_COORD attributes for encoding.

```C++
std::vector<unsigned int> faces{4, 6, 0, 6, 4, 5, 2, 6, 3, 3, 5, 2, 1, 2, 5,
                                  0, 6, 2, 0, 1, 4, 1, 5, 4, 3, 6, 5, 0, 2, 1};
std::vector<float> vertices{
    0.0097742,  0.05386876,  -0.00861827, -0.00014776, 0.02546649,
    -0.0237071, -0.03712369, 0.012704,    -0.00307178, -0.03941451,
    0.00185751, -0.03519752, 0.03155411,  0.00345094,  -0.00537058,
    0.01412319, -0.00706513, -0.03565174, 0.01980647,  0.05217797,
    -0.03568564};

std::vector<float> text{6, 0, 1, 2, 3, 4, 5};
  
auto mesh = std::unique_ptr<draco::Mesh>(new draco::Mesh());
mesh->SetNumFaces(faces.size() / 3);
mesh->set_num_points(vertices.size() / 3);

for (std::size_t i = 0; i <= faces.size() - 3; i += 3) {
  mesh->SetFace(draco::FaceIndex(i / 3), {{draco::PointIndex(faces[i]),
                                           draco::PointIndex(faces[i + 1]),
                                           draco::PointIndex(faces[i + 2])}});
}

draco::GeometryAttribute va;
va.Init(draco::GeometryAttribute::POSITION, nullptr, 3, draco::DT_FLOAT32, false, draco::DataTypeLength(draco::DT_FLOAT32) * 3, 0);
std::unique_ptr<draco::PointAttribute> pa = std::unique_ptr<draco::PointAttribute>(new draco::PointAttribute(va));
pa->SetIdentityMapping();
pa->Reset(vertices.size() / 3);
for (int i = 0; i <= vertices.size() - 3; i += 3) {
  pa->SetAttributeValue(draco::AttributeValueIndex(i / 3), draco::VectorD<float, 3>(vertices[i], vertices[i + 1], vertices[i + 2]).data());
}
int att_id = mesh->AddAttribute(std::move(pa));
mesh->SetAttributeElementType(att_id, draco::MESH_CORNER_ATTRIBUTE);

draco::GeometryAttribute va2;
va2.Init(draco::GeometryAttribute::TEX_COORD, nullptr, 1, draco::DT_FLOAT32, false, draco::DataTypeLength(draco::DT_FLOAT32), 0);
std::unique_ptr<draco::PointAttribute> pa2 = std::unique_ptr<draco::PointAttribute>(new draco::PointAttribute(va2));
pa2->SetIdentityMapping();
pa2->Reset(text.size());
for (int i = 0; i < text.size(); i++) {
  float v = text[i];
  pa2->SetAttributeValue(draco::AttributeValueIndex(i), (void *)&v);
}
att_id = mesh->AddAttribute(std::move(pa2));
mesh->SetAttributeElementType(att_id, draco::MESH_CORNER_ATTRIBUTE);

mesh->DeduplicateAttributeValues();
mesh->DeduplicatePointIds();

draco::Mesh *mesh_raw = mesh.get();

draco::Encoder encoder;
draco::DecoderBuffer decoderBuffer;
draco::Decoder decoder;

draco::EncoderBuffer buffer1;
encoder.SetSpeedOptions(4, 4);
encoder.SetAttributeQuantization(draco::GeometryAttribute::POSITION, 10);
encoder.SetAttributeQuantization(draco::GeometryAttribute::TEX_COORD, 10);
auto status1 = encoder.EncodeMeshToBuffer(*mesh_raw, &buffer1);
decoderBuffer.Init(buffer1.data(), buffer1.size());
auto statusor1 = decoder.DecodeMeshFromBuffer(&decoderBuffer);
std::cout << "status1: " << status1.code() << std::endl;          //OK
std::cout << "statusor1.ok(): " << statusor1.ok() << std::endl;   //OK

draco::EncoderBuffer buffer2;
encoder.SetSpeedOptions(4, 4);
encoder.SetAttributeQuantization(draco::GeometryAttribute::POSITION, 10);
//encoder.SetAttributeQuantization(draco::GeometryAttribute::TEX_COORD, 10);
auto status2 = encoder.EncodeMeshToBuffer(*mesh_raw, &buffer2);
decoderBuffer.Init(buffer2.data(), buffer2.size());
auto statusor2 = decoder.DecodeMeshFromBuffer(&decoderBuffer);
std::cout << "status2: " << status2.code() << std::endl;          //OK
std::cout << "statusor2.ok(): " << statusor2.ok() << std::endl;   //OK

draco::EncoderBuffer buffer3;
encoder.SetSpeedOptions(3, 3);
encoder.SetAttributeQuantization(draco::GeometryAttribute::POSITION, 10);
encoder.SetAttributeQuantization(draco::GeometryAttribute::TEX_COORD, 10);
auto status3 = encoder.EncodeMeshToBuffer(*mesh_raw, &buffer3);
decoderBuffer.Init(buffer3.data(), buffer3.size());
auto statusor3 = decoder.DecodeMeshFromBuffer(&decoderBuffer);
std::cout << "status3: " << status3.code() << std::endl;          //OK
std::cout << "statusor3.ok(): " << statusor3.ok() << std::endl;   //FAILURE
```

**Comment by ondys** [comment]
Hm I believe the issues is that with speed <=3, we use predictor for texture coordinates that expect the texture coordinates to have 2 components, while in your example you define texture coordinates with a single component. We should certainly not use that predictor in this case but do you actually want to use texture coordinates with one component?

By the way, it may be easier to initialize the attributes using PointAttribute::Init method instead of going through the GeometryAttribute. E.g.:
```
std::unique_ptr<PointAttribute> pa(new PointAttribute());
pa->Init(draco::GeometryAttribute::POSITION, 3, draco::DT_FLOAT32, false, vertices.size() / 3);
for (int i = 0; i <= vertices.size() - 3; i += 3) {
  pa->SetAttributeValue(draco::AttributeValueIndex(i / 3), draco::VectorD<float, 3>(vertices[i], vertices[i + 1], vertices[i + 2]).data());
}
```

**Comment by ondys** [comment]
The issue should be fixed  in the latest release (1.5.2)
